subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java

```diff
@@ -1513,19 +1518,73 @@ public boolean isAllowExtraLogging() {
 
     @Override
     public GradleExecuter startBuildProcessInDebugger(boolean flag) {
-        debug = flag;
+        debug.getEnabled().set(flag);
         return this;
     }
 
+
+    @Override
+    public GradleExecuter startBuildProcessInDebugger(boolean flag, boolean server, boolean suspend, int port, String host) {
```

Comment:
```suggestion
    public GradleExecuter startBuildProcessInDebugger(boolean flag, Action<DebugOptions> debugOptions) {
```

🤔 Maybe an API like this would be better? Then you could configure the debug options in a lambda/Closure. For Groovy, we could also accept a `Closure` here with `@DelegatesTo` which would allow an even better API.